File: Exercises/101-109 - Small Problems/easy_3/palindromic_strings_1.rb
# Write a method that returns true if the string passed as an argument is a palindrome, false otherwise.
# (A palindrome reads the same forwards and backwards.) Case matters, and all characters should be considered.

def palindrome?(string)
  string == string.reverse
end

puts palindrome?('madam') == true
puts palindrome?('Madam') == false          # (case matters)
puts palindrome?("madam i'm adam") == false # (all characters matter)
puts palindrome?('356653') == true

File: Exercises/210 - Small Problems/easy-3/palindromic-strings-1.js
/*
Write a function that returns true if the string passed as an argument is a palindrome, false otherwise. A palindrome
reads the same forwards and backwards. For this problem the case matters and all characters matter.

Examples:

isPalindrome('madam');                    // true
isPalindrome('Madam');                    // false (case matters)
isPalindrome('madam i\'m adam');          // false (all characters matter)
isPalindrome('356653');                   // true
*/

function isPalindrome(str) {
  return str === str.split('').reverse().join('');
}

console.log(isPalindrome('madam'));
console.log(isPalindrome('Madam'));
console.log(isPalindrome('madam i\'m adam'));
console.log(isPalindrome('356653'));